Atlassian.Jira/Remote/JiraRestClient.cs

```diff
@@ -242,5 +246,43 @@ private JToken GetValidJsonFromResponse(IRestRequest request, IRestResponse resp
                 return parsedContent;
             }
         }
+
+        private async Task<string> PrintResponseAsync(IRestRequest request)
+        {
+            try
+            {
+                using var handler = new HttpClientHandler()
+                {
+                    AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate
+                };
+                handler.ServerCertificateCustomValidationCallback += (sender, certificate, chain, errors) => true;
+                HttpClient.DefaultProxy = _restClient.Proxy;
+                var client = new HttpClient(handler);
+
+                var authHeader = request.Parameters.FirstOrDefault(header => header.Name == "Authorization");
+                var splitAuthHeader = authHeader?.Value?.ToString()?.Split();
+
+                if (splitAuthHeader == null || splitAuthHeader.Length < 2)
+                {
+                    return null;
+                }
+
+                var authValue = splitAuthHeader[1];
+
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", authValue);
+                client.DefaultRequestHeaders.Add("Connection", "Keep-Alive");
+                client.DefaultRequestHeaders.Add("Accept", "application/json, text/json, text/x-json, text/javascript, application/xml, text/xml");
+                client.DefaultRequestHeaders.Add("User-Agent", "RestSharp/106.11.7.0");
+                var response = await client.GetAsync($"{_restClient.BaseUrl}{request.Resource}");
+                var headers = response.Headers;
+                return $"Retry request attempt result = {Convert.ToBase64String(Encoding.ASCII.GetBytes(headers.ToString()))}";
+            }
+            catch (Exception exception)
+            {
+                Trace.WriteLine("Failed to retry request");

```

Comment:
Why not return this message and the ex details?

Reply:
didn't want to spam, but sure, adding 👍